Actions/CheckForNewWarnings/CheckForNewWarnings.ps1

```diff
@@ -0,0 +1,44 @@
+Param(
+    [string] $token,
+    [string] $project,
+    [object] $settings,
+    [string] $targetBranch,
+    [string] $prBuildOutputFile
+)
+
+if (-not $ENV:GITHUB_BASE_REF)
+{
+    Write-Host "Checking for warnings only runs on pull requests."
+    return
+}
+
+if (-not $settings.checkForNewWarnings)
+{
+    Write-Host "Checking for new warnings is disabled in the settings."
+    return
+}
+
+Write-Host "Analyzing build for new warnings..."
```

Comment:
```suggestion
Write-Host "::group::Analyzing build for new warnings..."
```
Consider doing something like this (and add an Write-Host "::endgroup::" at the bottom of the script). It'll help organize the GitHub log 
![image](https://github.com/user-attachments/assets/6d59019e-377f-49a6-a5ba-6e1fee3a8ab3)